src/node/txreconciliation.cpp

```diff
@@ -28,6 +28,34 @@ static uint256 ComputeSalt(uint64_t local_salt, uint64_t remote_salt)
     return (CHashWriter(RECON_SALT_HASHER) << salt1 << salt2).GetSHA256();
 }
 
+/**
+ * Used to keep track of the ongoing reconciliations, the transactions we want to announce to the
+ * peer when next transaction reconciliation happens, and also all parameters required to perform
+ * reconciliations.
+ */
+class ReconciliationState {
+
+    public:
+
+    /**
+     * Reconciliation involves exchanging sketches, which efficiently represent transactions each
+     * peer wants to announce. Sketches are computed over transaction short IDs.
+     * These values are used to salt short IDs.
+     * TODO: These are currently public to overcome compiler warning (-Wunused-private-field).
+     *       They should be made private once they are used in one of the following commits.
+     */
+    const uint64_t m_k0, m_k1;
+
+    /**
+     * Reconciliation protocol assumes using one role consistently: either a reconciliation
+     * initiator (requesting sketches), or responder (sending sketches). This defines our role.
+     * */
+    const bool m_we_initiate;
```

Comment:
Yeah my initial intention was to make it good for future where both could be true or false, and avoid having a 4-way variant of {YESYES NONO YESNO NOYES}.
But hey, since this future might be very far ahead, you're probably right. Will do.

Reply:
Actually, I think, it would be very useful if you take a look at the [final Erlay code fragment](https://github.com/bitcoin/bitcoin/blob/5728fac4d3d29f64ea811f5978e80dabdc083d87/src/txreconciliation.cpp#L230)?

So, the structure there is:
1. class `ReconciliationState`, which has 2 fields: `ReconciliationInitByUs` and `ReconciliationInitByUs`.
2. Those two fields are structs, corresponding to Initiator and Responder roles. They encapsulate specific stuff.
3. The rest of `ReconciliationState` is stuff which is not specific to a role (salt, for example).

I guess I just used another pattern for that.

If I understand you correctly, you suggesting using std::variant<Initiator, Responder> instead of ReconciliationState. 

Is std::variant still a good candidate in that case? What would be the right way for them to have many fields in common?
I guess `Initiator` and `Responder` could be inherited from one shared parent with all that stuff.

Reply:
Sorry, I have realized I was missing some context. Right now, one peer is Initiator and the other is Responder, but do you envision it being more dynamic in future versions of the protocol? If so, perhaps it doesn't make sense to use a variant. But perhaps, since you only have 2 possibilities right now, you only need 1 bool instead of 2?

Reply:
Based on more discussions with Gloria, I think we keep this field, but in latter commits we switch to
`std::variant<ReconciliationInitByUs, ReconciliationInitByThem> m_current_recon;`
which would dictate us the role of the peer.